retrievalmarket/impl/provider.go

```diff
@@ -79,40 +87,48 @@ func NewProvider(minerAddress address.Address, node retrievalmarket.RetrievalPro
 		StateEntryFuncs: providerstates.ProviderStateEntryFuncs,
 		Notifier:        p.notifySubscribers,
 	})
-	p.stateMachines = statemachines
 	if err != nil {
 		return nil, err
 	}
+	p.Configure(opts...)
+	p.stateMachines = statemachines
 	return p, nil
 }
 
+func (p *Provider) RunDealDecisioningLogic(ctx context.Context, state retrievalmarket.ProviderDealState) (bool, string, error) {
+	if p.dealDecider == nil {
+		return true, "", nil
+	}
+	return p.dealDecider(ctx, state)
+}
+
 // Stop stops handling incoming requests
-func (p *provider) Stop() error {
+func (p *Provider) Stop() error {
```

Comment:
this had to be exported similarly to storagemarket, so a decider func is possible